### PR TITLE
Dismiss splash screen after initial `Window.Activate`, themed `RootVisual` background

### LINKED ISSUE #6393: Basic background does not take into account the active theme
## Current behavior

Differences(In WebAssembly relative to UWP):
The background is different unless explicitly set `Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"`. When testing, the system has a Dark theme installed.

## How to reproduce it (as minimally and precisely as possible)

https://github.com/HavenDV/Uno.WASM.TemplateBug
https://github.com/HavenDV/Uno.WASM.TemplateBug/commit/b6a4c8ed81b38ced515ba79add7849c00d6191d4

## Works on UWP/WinUI
Yes

## Environment

Nuget Package:
- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia>

Nuget Package Version(s):
- 3.8.6 and 3.9.0-dev.104

Affected platform(s):
Possible multiple, tested only on WebAssembly

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: latest)
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### LINKED ISSUE #10820: Implicit `Window` background
## What would you like to be added:

By default WinUI has a theme-aware background (black or white). This is not the case for Uno

## Why is this needed:

Consistency

## For which Platform:

- [ ] WebAssembly
- [ ] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): part of #10178, closes #10820, closes #6393

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

White background flickers if Window is activated with a delay.

## What is the new behavior?

In tandem with https://github.com/unoplatform/Uno.Wasm.Bootstrap/pull/699 this should ensure no flickering of white color will show up on Android, iOS, macOS and WASM. This does not help on Skia yet unfortunately, as there is no splash screen yet.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
The PR does not depend on the Bootstrapper changes and can be merged separately

File: src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_RootVisual.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	public async Task When_Theme_Changes()
+	{
+		var rootVisual = Uno.UI.Xaml.Core.CoreServices.Instance.MainRootVisual;
+		if (rootVisual is null)
+		{
+			// Ignore on Uno Islands
+			return;
+		}
+
+		Assert.AreEqual(Colors.White, (rootVisual.Background as SolidColorBrush).Color);
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_RootVisual.cs:29]:
```suggestion
		Assert.AreEqual(Colors.White, ((SolidColorBrush)rootVisual.Background).Color);
```

File: src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_RootVisual.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+		using (ThemeHelper.UseDarkTheme())
+		{
+			await TestServices.WindowHelper.WaitForIdle();
+			Assert.AreEqual(Colors.Black, (rootVisual.Background as SolidColorBrush).Color);
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_RootVisual.cs:34]:
```suggestion
			Assert.AreEqual(Colors.Black, ((SolidColorBrush)rootVisual.Background).Color);
```

File: src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_RootVisual.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		}
+
+		await TestServices.WindowHelper.WaitForIdle();
+		Assert.AreEqual(Colors.White, (rootVisual.Background as SolidColorBrush).Color);
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_RootVisual.cs:38]:
```suggestion
		Assert.AreEqual(Colors.White, ((SolidColorBrush)rootVisual.Background).Color);
```

File: src/Uno.UI/UI/Xaml/Window.cs [hunk 4]
```diff
@@ -200,16 +216,16 @@ public void Activate()
...
 			_current ??= this;
-
-			InternalActivate();
-
-			OnActivated(CoreWindowActivationState.CodeActivated);
+			_wasEverActivated = true;
 
 			// Initialize visibility on first activation.
 			Visible = true;
+			OnActivated(CoreWindowActivationState.CodeActivated);
```
Comment (nickrandolph) [inline src/Uno.UI/UI/Xaml/Window.cs:223]: This feels the wrong way around - I would expect the platform activation (ie ActivatePartial) to happen first, before the public Activated event is raised (which is inside the OnActivated method)

File: src/Uno.UI/UI/Xaml/Window.cs [hunk 5]
```diff
@@ -255,19 +277,12 @@ internal void OnActivated(CoreWindowActivationState state)
...
-					this.Log().LogDebug($"Window visibility changing to {newVisibility}");
-				}
-
-				Visible = newVisibility;
-
-				var args = new VisibilityChangedEventArgs() { Visible = newVisibility };
-				CoreWindow.OnVisibilityChanged(args);
-				VisibilityChanged?.Invoke(this, args);
+				return;
 			}
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Window.cs:283]: It may be interesting to keep the debug message ?

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/ts/WindowManager.ts [hunk 6]
```diff
@@ -57,6 +57,16 @@ namespace Uno.UI {
 		 * */
 		private static buildSplashScreen(): Promise<boolean> {
 			return new Promise<boolean>(resolve => {
+
+				let bootstrapperLoaders = document.getElementsByClassName("persistent-loader");
```
Comment (carldebilly) [inline src/Uno.UI/ts/WindowManager.ts:61]: Should we prefix our class names with something uno specific?

**Review by carldebilly — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Window.cs (hunk 5 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Window.cs:283]: The log is still there, just moved to the property setter 👍

**Pull request merged**